### PR TITLE
docs: switch docs engine

### LINKED ISSUE #457: docs: move from jekyll pages to a better doc engine
Items;
- [ ] change typedoc to generate to the new temple engine
- [ ] change the deploy config on gh-pages
- [ ] make a reusable block for `### Full SDK Installation` and reuse them inside packages `README` file

Notes;
- One good template is the [nextra](https://nextra.vercel.app/).

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close #457

### CODE REVIEW

File: packages/apps/docs/.vitepress/config.ts
Comment (arboleya) [inline packages/apps/docs/.vitepress/config.ts:9]: Amazing. 👏 

One of our simplest yet best doc features is the `See Code in Context` link we currently show after every imported code snippet. I can't repeat enough how useful I find that to be. The fact no one has implemented it in any tool only shows how accurate @camsjams was with that Typedoc plugin, and I'm sure this will be the final touch for all other snippets goodies Vitepress provides.

Good job keeping it alive via this `addContextLinkPlugin`!

**Comment by Torres-ssf** [comment]
Since the new docs build/deploy process is going to change, I am assuming we won't need these Ci steps anymore
[1](https://github.com/FuelLabs/fuels-ts/blob/dd7b1cab0e7c4a5234383ce6fc34f041ee6d03a9/.github/workflows/release.yaml#L48)
[2](https://github.com/FuelLabs/fuels-ts/blob/dd7b1cab0e7c4a5234383ce6fc34f041ee6d03a9/scripts/changeset-version-with-docs.ts#L1-L32)
>cc @camsjams @Dhaiwat10 @arboleya

**Comment by Dhaiwat10** [comment]
> Since the new docs build/deploy process is going to change, I am assuming we won't need these Ci steps anymore [1](https://github.com/FuelLabs/fuels-ts/blob/dd7b1cab0e7c4a5234383ce6fc34f041ee6d03a9/.github/workflows/release.yaml#L48) [2](https://github.com/FuelLabs/fuels-ts/blob/dd7b1cab0e7c4a5234383ce6fc34f041ee6d03a9/scripts/changeset-version-with-docs.ts#L1-L32)
> 
> > cc @camsjams @Dhaiwat10 @arboleya

Yeah, if we are not using `typedoc` and building our docs like that, I believe we can get rid of these actions.

**Comment by Dhaiwat10** [comment]
Re. your previous comment @Torres-ssf, I will pretty much second what @arboleya said. Let's go with Algolia, and I feel like the general consensus is to move on from `typedoc` too.

File: apps/docs/.vitepress/plugins/snippetPlugin.ts
Comment (arboleya) [inline apps/docs/.vitepress/plugins/snippetPlugin.ts:89]:
```suggestion
        snippetRegion = '',
```

**Review by arboleya — CHANGES_REQUESTED** [review]
I second what @camsjams said about fixing the history for the `*.md` files.

To fix it with a single commit, you should do something like this:

 1. First, make a `copy` of your current repo/dir locally
 1. Switch to `master` and make sure it's up to date:
    - `git checkout master`
    - `git pull origin master`
 1. Delete the current `branch`:
    - `git branch -D torres/docs/new-docs-engine`
 1. Start a new branch from `master` using the same name:
    - `git checkout -b torres/docs/new-docs-engine`
 3. Copy everything from your `copy` _(step 1)_ into it:
    - _Except the `.git` dir_
 5. Commit everything in a single commit:
    - `git add . && git commit -m 'Switching docs engine'`
 6. Force push it:
    - `git push origin torres/docs/new-docs-engine --force`

Conclusion:
 - Your commits (from this PR) will be lost
 - The file's history will be kept

Perhaps there are other ways to achieve this and keep your commits.

File: .github/workflows/release.yaml
Comment (Torres-ssf) [reply .github/workflows/release.yaml:80]: @arboleya I tried the suggested approach while running some tests on previous commits and it resulted in an error. The current one seems to be working.

**Comment by Torres-ssf** [comment]
> @Torres-ssf The `*.md` files are showing as added instead renamed; you forgot to delete the _[old]_ docs directory. You'll need to delete it and then do a `--amend` commit followed by another `push --force`. Something like this.

@arboleya Yeah, you were right. I did forget to delete the old docs. It's done now. Everything on the same commit

**Comment by arboleya** [comment]
Sweet.
sweet

Bitter.
bitter

Jokes aside, I'll give it a good review later.

File: apps/docs/src/versions.data.ts
Comment (arboleya) [inline apps/docs/src/versions.data.ts:54]: I don't think you need the `sway` variable, `forc` can be used everywhere.

File: packages/versions/package.json
Comment (arboleya) [inline packages/versions/package.json:14]: Are you still using the `@fuel-ts/versions` to fetch the versions?

If not, perhaps this workaround can be removed before merging.

Also because of #827

**Review by arboleya — CHANGES_REQUESTED** [review]
Great job! 👏 

I left some minor observations.

File: apps/docs/src/guide/testing/index.md
Comment (arboleya) [reply apps/docs/src/guide/testing/index.md:5]: We use [MDX](https://mdxjs.com/table-of-components/), no?

Anyway, I was hoping to get something similar to this:

```ts
// file.mdx?
import { FORC } from './versions.ts'

<a url="http://domain.com/{{version}}" />
```

Comment (Torres-ssf) [reply apps/docs/src/guide/testing/index.md]: @arboleya No, Vitepress does not support MDX files. What it supports instead is using Vue components inside the `.md` files.

Comment (arboleya) [reply apps/docs/src/guide/testing/index.md]: Ok, I misunderstood it then.

A `<Component />` might help make it less repetitive. If the idea makes sense, please open an issue so we can discuss it further.

**Review by arboleya — APPROVED** [review]
Great work. 👏 

Some numbers I measured:
 - Startup time: `~2s`
 - Build time: `~5s`

File: packages/fuel-gauge/src/contract.test.ts
Comment (Torres-ssf) [reply packages/fuel-gauge/src/contract.test.ts:706]: The Vitepress native import script plugin works this way so when implementing a custom one this was one thing that I did not change.

File: apps/docs/src/versions.data.ts
Comment (arboleya) [reply apps/docs/src/versions.data.ts:43]: @camsjams Initially, the `versions` package would return `0.0.0` for everything when developing, and then the CI would inject the proper versions during the release. IIRC, this was a problem for the Docs

@Torres-ssf Was it for showing the proper links or something like that? Can't remember it precisely; perhaps I'm confused. Either way, I'll have to refactor the `versions` package to fix a problem that arose after #827. In the end, it might be suitable for such use.

File: packages/fuel-gauge/src/contract.test.ts
Comment (Torres-ssf) [reply packages/fuel-gauge/src/contract.test.ts:706]: @camsjams I think the main reason behind it is because it supports inner scripts.

Screenshot from 2023-03-20 20-54-31
Screenshot from 2023-03-20 20-32-03

File: apps/docs/src/versions.data.ts
Comment (Torres-ssf) [reply apps/docs/src/versions.data.ts:43]: Exactly @arboleya . It was made this way for both reasons.

Comment (arboleya) [reply apps/docs/src/versions.data.ts:43]: Cool, I'll let you know how this pan out.

**Review by arboleya — DISMISSED** [review]

**Comment by Torres-ssf** [comment]
Before merging this to master we need to first change the project GitHub pages settings.

Change this:
Screenshot from 2023-03-21 09-54-57

Into `GitHub actions`:
Screenshot from 2023-03-21 09-55-29

File: apps/docs/src/versions.data.ts
Comment (arboleya) [reply apps/docs/src/versions.data.ts:43]: Follow up [issue](https://github.com/FuelLabs/fuels-ts/issues/849) and [PR](https://github.com/FuelLabs/fuels-ts/pull/850).

**Review by arboleya — DISMISSED** [review]

**Review by arboleya — DISMISSED** [review]

File: .changeset/heavy-clouds-arrive.md
Comment (Dhaiwat10) [inline .changeset/heavy-clouds-arrive.md:1]: I noticed three of these empty changeset files. Is this intentional?

File: apps/docs/.vitepress/config.ts
Comment (Dhaiwat10) [inline apps/docs/.vitepress/config.ts:7]:
```suggestion
  description: 'Fuel Labs Typescript SDK',
```

Sorry for the nitpicking 😅

File: apps/docs/.vitepress/plugins/codeInContextPlugin.ts
Comment (Dhaiwat10) [inline apps/docs/.vitepress/plugins/codeInContextPlugin.ts:25]: Clean!

**Review by Dhaiwat10 — APPROVED** [review]
Amazing work @Torres-ssf! Left some minor non-blocking comments.

**Comment by Torres-ssf** [comment]
@arboleya @Dhaiwat10 Have worked on all requested changes and need the approval again :sweat_smile:

**Review by arboleya — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**
